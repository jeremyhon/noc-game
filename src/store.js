import { createStore } from 'redux'
import rootReducer from './reducer'

const startStore = (initialState) => {
  return createStore(
    rootReducer,
    initialState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
}

const store = startStore()

export default store
