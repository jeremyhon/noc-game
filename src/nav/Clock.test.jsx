import React from 'react'
import { shallow } from 'enzyme'

import store from '../store'
import ConnectedClock, { Clock } from './Clock'

it("matches the snapshot", () => {
  expect(shallow(<Clock dayCount={1} />)).toMatchSnapshot()
})

it("receives the day count from the store correctly", () => {
  expect(shallow(<ConnectedClock store={store} />)).toMatchSnapshot()
})
