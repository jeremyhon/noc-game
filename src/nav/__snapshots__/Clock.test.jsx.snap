// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<span>
  Y: 
  1
   M: 
  1
   D: 
  2
</span>
`;

exports[`receives the day count from the store correctly 1`] = `
<Clock
  dayCount={0}
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
